genai-perf/genai_perf/inputs/converters/base_converter.py

```diff
@@ -52,6 +52,20 @@ def _select_model_name(self, config: InputsConfig, index: int) -> str:
                 f"Model selection strategy '{config.model_selection_strategy}' is unsupported"
             )
 
+    def _construct_text_payload_batch_agnostic(
```

Comment:
Note that we currently return different schemas in the retrievers when client-side batching is enabled (batch_size >= 2), which is why this function is necessary.

In a future refactor ticket, I may look at using the same schema (Dicts/Lists) for both. Tthat would need to be done after a redesign of image client-side batching. I think the current client-side batching approach batches images and texts equally, so we'd need to disambiguate the two first. Happy to have this conversation offline.